server/api/api.go

```diff
@@ -114,12 +118,85 @@ func (h *Handler) setLink(w http.ResponseWriter, r *http.Request) {
 	status := http.StatusNotModified
 	if changed {
 		if err := h.store.SaveLinks(links); err != nil {
-			h.handleError(w, errors.Wrap(err, "unable to save link"))
+			h.handleError(w, errors.Wrap(err, "unable to save the link"))
 			return
 		}
 		status = http.StatusOK
 	}
 
+	ReturnStatusOK(status, w)
+}
+
+func (h *Handler) deleteLink(w http.ResponseWriter, r *http.Request) {
+	autolinkName := r.URL.Query().Get(autolinkclient.AutolinkNameQueryParam)
+	if autolinkName == "" {
+		h.handleError(w, errors.New("autolink name or pattern should not be empty"))
+		return
+	}
+
+	links := h.store.GetLinks()
+	found := false
+	for i := 0; i < len(links); i++ {
+		if links[i].Name == autolinkName {
+			links = append(links[:i], links[i+1:]...)
+			found = true
```

Comment:
As there is only one possible match here, we can break on found:

```suggestion
			found = true
			break
```